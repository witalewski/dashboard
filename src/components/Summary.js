import React from 'react';
import styled from '@emotion/styled';

import { format } from 'date-fns';
import { HorizontalBar } from './charts/HorizontalBar';

const StyledSummary = styled.div`
  width: 100%;
  height: 100%;

  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;

  .main-label {
    font-weight: bold;
  }

  .main-sum {
    font-weight: bold;
    font-size: 36px;
  }

  .container--stretch {
    align-self: stretch;
  }

  .comparison {
    display: flex;
    justify-content: space-between;
  }

  .comparison__sum {
    font-weight: bold;
    font-size: 24px;
  }

  .percent-of-average-label {
    text-align: center;
    padding-bottom: 24px;
  }
`;

export const Summary = ({ data }) => (
  <StyledSummary>
    <div>
      <div className="main-label">In {format(new Date(), 'MMMM')} you did</div>
      <div className="main-sum">${data.revenueThisMonth}</div>
    </div>

    <div className="container--stretch">
      <div className="comparison">
        <div className="comparison__label">Last month you did:</div>
        <div className="comparison__sum">${data.revenueLastMonth}</div>
      </div>
      <div className="comparison">
        <div className="comparison__label">Average monthly revenue:</div>
        <div className="comparison__sum">${data.revenueAverage}</div>
      </div>
    </div>

    <div className="container--stretch">
      <HorizontalBar percent={data.revenueRate * 100} />
      <div className="percent-of-average-label">
        of your monthly average revenue
      </div>
    </div>
  </StyledSummary>
);
