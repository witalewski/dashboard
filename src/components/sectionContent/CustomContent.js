import React from 'react';
import styled from '@emotion/styled';

const CustomContentStyled = styled.div`
  width: 100%;
  height: 100%;
  min-width: 200px;
`;

export const CustomContent = ({ children }) => (
  <CustomContentStyled>{children}</CustomContentStyled>
);
