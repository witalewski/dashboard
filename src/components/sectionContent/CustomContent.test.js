import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import mockChartData from '../../fixtures/mockChartData.json';
import { CustomContent } from './CustomContent';
import { Summary } from '../Summary';

describe('CustomContent', () => {
  it('matches snapshot with mock data', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <CustomContent className="summary-section-wrapper" title="Summary">
        <Summary data={mockChartData.summaryChart.data} />
      </CustomContent>
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
