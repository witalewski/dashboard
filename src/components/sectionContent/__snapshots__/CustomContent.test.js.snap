// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomContent matches snapshot with mock data 1`] = `
<ForwardRef(render)>
  <Summary
    data={
      Object {
        "revenueAllTime": 262.22,
        "revenueAverage": 131.11,
        "revenueLastMonth": 199.09,
        "revenueRate": 0.38,
        "revenueThisMonth": 100.64,
      }
    }
  />
</ForwardRef(render)>
`;
